src/components/AssetIcon.tsx

```diff
@@ -118,14 +117,48 @@ export const AssetIcon: React.FC<AssetIconProps> = ({
 
   // For Soroban custom tokens, use the Soroban logo
   if (token.type === AssetTypeWithCustomToken.CUSTOM_TOKEN) {
+    const tokenIdentifier = getTokenIdentifier(token);
+    let icon = icons[tokenIdentifier];
```

Comment:
I believe our `icons` store currently only store icons for Classic Assets which means it'll never return an icon for Soroban Assets.